src/main/resources/templates/db_scripts/schema.json

```diff
@@ -922,7 +922,7 @@
     {
       "run": "after",
       "snippetPath": "oaipmh/createOaiPmhViewFunction.sql",
-      "fromModuleVersion": "19.2.4"
+      "fromModuleVersion": "19.2.5"
```

Comment:
Should it be 19.3.0?

Reply:
Maybe, we need to make a new release, but I doubt it will be 19.3.0

Reply:
hmm, but the PR is created to `master`, and it is already of 19.3.0 version.

Reply:
I think it will be a hotfix release, nevertheless this change is needed to be in master too.

Reply:
There was a recent workaround where a bug fix version was used in a `fromModuleVersion` in master in order to avoid the same script being executed twice if the tenant had already upgraded to a bug fix version already containing the change.

Personally, I don't think this is a good regular pattern, I think the mainline code should refer to versions that are / could be present on the mainline.

Given that this change breaks compatibility, the module version should be changed to 20.0.0 and I think that is what should be present here as well.

If this breaking change is to be back ported (which I strongly disagree with, however I'm aware that exceptional permission has been granted for this) then I think the `fromModuleVersion` should be changed in the bug fix branch.

That said, if this script is destructive or non-repeatable, and folks want to apply the same workaround, then this version should be the version *after* the bug fix version that contained it. 

@julianladisch Is that a correct understand of how these versions work and how the workaround works?

Reply:
As I understand correctly this is the question of duscussion now which version to leave.

Reply:
I left the version 19.2.5 because latest changes made with mind of backward compatible. To all change shouldn't lead to breakdown of former functionality.

Reply:
> hmm, but the PR is created to `master`, and it is already of 19.3.0 version.

master is on 19.3.0-SNAPSHOT.

FOLIO by default uses the next feature version as the *-SNAPSHOT development version for master: https://dev.folio.org/guidelines/release-procedures/#prepare-and-perform-the-source-release

When doing the next release the maintainer needs to adjust the version: If there are only bugfixes use v19.2.5. If there are new features use v19.3.0. If there are breaking changes use v20.0.0.

The 19.3.0-SNAPSHOT on master does not reflect whether the commits that were made after the last release are bugfixes, features or breaking changes. This is postponed until the next version gets actually released.

Reply:
`fromModuleVersion` is the version where the script has been introduced: https://github.com/folio-org/raml-module-builder#the-post-tenant-api
19.2.5 is correct if we release this pull request as 19.2.5.

Reply:
> The 19.3.0-SNAPSHOT on master does not reflect whether the commits that were made after the last release are bugfixes, features or breaking changes. This is postponed until the next version gets actually released.

As the next mainline version is always intended to be the next minor version, I tend to consider compatible feature changes and bug fixes as not affecting the version. 

I tend to try to identify compatibility breaking changes during the pull request and change the next version appropriately within that pull request. Otherwise what happens is that SNAPSHOT versions of a module version are published that do not reflect the intent of the changes.

This is less problematic for modules, mostly only misleading, it can be significant for libraries.

As a lead maintainer, I want the defined version to be accurate to what should be next released when I come to make the formal release. Otherwise, I have to know enough about all of the changes in order to evaluate that myself, which can be difficult and time consuming when that is 3 months of changes made by multiple teams autonomously.

I do tend to do some checks to spot unacknowledged compatibility breaking changes. It isn't unusual for those to have been made without the version changing.

Reply:
This pull request is no longer a breaking change. But it adds two new API endpoints: `oai-pmh-view/updatedInstanceIds`, `/oai-pmh-view/enrichedInstances`. This indicates that the release version should be 19.3.0 to signal that it is a feature version. I understand the hot-fix releases should be restricted to bug fixes, but this pull request qualifies as an exceptions to this rule because it has very low risk.

Reply:
> 19.2.5 is correct if we release this pull request as 19.2.5.

This pull request is intended to be merged to the mainline. To me, that means that this change is actively intended for 19.3.0.

If folks decide to also port these changes to 19.2.5 then that will be a separate pull request, with different intentions. (I won't repeat my concerns about that approach here)

This is why I personally think the `fromVersion` in this pull request should be `19.3.0` unless we are aiming to apply a similar workaround to avoid double application of scripts when upgrading from a bug fix release.

Reply:
> This indicates that the release version should be 19.3.0 to signal that it is a feature version. I understand the hot-fix releases should be restricted to bug fixes, but this pull request qualifies as an exceptions to this rule because it has very low risk.

I believe the mainline version is already 19.3.0, meaning that this pull request does not need to actively change the version.

Reply:
Thank you for clarification, so the version 19.2.5 will be retained which doesn't forbid to include thesechanged to the future 19.3.0 from main buildline.